internal/clusterinfo/clusterinfo.go

```diff
@@ -0,0 +1,118 @@
+package clusterinfo
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	k8sConfigMap string = "kyma-cluster-info"
+	k8sNamespace string = "kube-system"
+)
+
+// ClusterInfo contains data about the current cluster
+type ClusterInfo struct {
+	k8sClient   kubernetes.Interface
+	initialized bool
+	isLocal     bool
+	provider    string
+}
+
+// NewClusterInfo creates a new cluster info instance
+func NewClusterInfo(k8sClient kubernetes.Interface) *ClusterInfo {
```

Comment:
If the package is called cluster info, then this function should be called `New()`.
Strange that the linter did not catch this, it is called "stuttering" in the Go ecosystem (because when used it would read like: `clusterinfo.NewClusterInfo()`).

Reply:
Good points.. will rename it.

Reply:
Fixed.